### DISCUSSION TITLE
How can I have the same access in development mode and production mode?

### DISCUSSION THREAD

**Discussion by chaos-zhu** [nuxt/nuxt#22339] [Questions] [answered]
### Describe the feature

I want to do something after the service starts, such as executing a scheduled task, starting another service, or something else.

While developing the pattern, I can use the `hooks: listen` implementation in `nuxt.config.ts`, but it won't work after `npm run build`.

I looked at the nitro document, i can use `presets` to define entry (https://nitro.unjs.io/deploy/custom-presets), but it only work in production mode. This is annoying, I can do this easily in nuxt2, is there a solution to implement it in nuxt3?

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by danielroe** [comment, marked as answer]
You can use a nitro plugin to do this.

**Reply by TheAlexLichter** [reply]
As an addition: The `listen` hook is clearly defined as called "when the dev server is loading"

https://github.com/nuxt/nuxt/blob/198f9a23ec7e091fa0aea2f089c4098c5bda9073/packages/schema/src/types/hooks.ts#L231-L236

As Daniel said, a Nitro plugin is the way to go 😋
